src/main/java/it/eng/idsa/businesslogic/processor/sender/SenderParseReceivedDataProcessorBodyBinary.java

```diff
@@ -72,14 +73,14 @@ public void process(Exchange exchange) throws Exception {
 
 	/**
 	 * Check if header content type is application/json; UTF-8 or application/json
+	 * 
 	 * @param contentType
 	 * @return
 	 */
 	private boolean checkHeaderContentType(String contentType) {
-		if (contentType != null && (contentType
-				.equals(ContentType.APPLICATION_JSON.toString())
+		if (ContentType.APPLICATION_JSON.toString().equals(contentType)
 				|| contentType.equals(ContentType.create("application/json").toString())
-				|| contentType.equals(ContentType.create("application/json+ld").toString()))) {
+				|| contentType.equals(ContentType.create("application/json+ld").toString())) {
```

Comment:
What about other checks?
contentType.equals(ContentType.create("application/json+ld").toString())
to
ContentType.create("application/json+ld").toString()).equals(contentType)
and application/json one?